### PR TITLE
[Instrumentation.AWS] Support .NET 6 and resolve AoT warning

### LINKED ISSUE #1543: AWSTracingPipelineHandler produces IL2075 warning when compiling for native AoT
# Issue with OpenTelemetry.Instrumentation.AWS

List of [all OpenTelemetry NuGet packages](https://www.nuget.org/profiles/OpenTelemetry) and version that you are using (e.g. `OpenTelemetry 1.3.2`):

* `OpenTelemetry.Instrumentation.AWS 1.1.0-beta.2`

Runtime version:

* `net6.0`+

**Is this a feature request or a bug?**

* [x] Feature Request
* [ ] Bug

**What is the expected behavior?**

When OpenTelemetry.Instrumentation.AWS is referenced in an application compiled with native AoT, there are no trim warnings.

**What is the actual behavior?**

1 trim warning is produced by the assembly's own code (an additional 2 are from the AWS SDK, see #1542).

```console
C:\Coding\open-telemetry\opentelemetry-dotnet-contrib\src\OpenTelemetry.Instrumentation.AWS\Implementation\AWSTracingPipelineHandler.cs(141): Trim analysis error IL2075: OpenTelemetry.Instrumentation.AWS.Implementation.AWSTracingPipelineHandler.AddRequestSpecificInformation(Activity,IRequestContext,String): 'this' argument does not satisfy 'DynamicallyAccessedMemberTypes.PublicProperties' in call to 'System.Type.GetProperty(String)'. The return value of method 'System.Object.GetType()' does not have matching annotations. The source value must declare at least the same requirements as those declared on the target location it is assigned to. [C:\Coding\open-telemetry\opentelemetry-dotnet-contrib\test\OpenTelemetry.AotCompatibility.TestApp\OpenTelemetry.AotCompatibility.TestApp.csproj]
```

This warning comes from this line:

https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/8a89a69445a1f89917a6a159e895b9267c7cc925/src/OpenTelemetry.Instrumentation.AWS/Implementation/AWSTracingPipelineHandler.cs#L137

## Additional Context

Found while going through the AWS-related libraries while looking for the source of trim warnings.

I did have a look at trying to fix this one myself, but it was beyond my expertise in this case.

### PR DESCRIPTION
**Pull request by martincostello**
Fixes #1543.

## Changes

- Add `net6.0` TFM and fix nullability warnings.
- Resolve native AoT warning.
- Mark `AWSTracingPipelineHandler` as `sealed` (e.g. #1510).
- Fix `IDE0090` suggestions.

* [x] Appropriate `CHANGELOG.md` updated for non-trivial changes

### CODE REVIEW

File: src/OpenTelemetry.Instrumentation.AWS/Implementation/AWSTracingPipelineHandler.cs [hunk 1]
```diff
@@ -128,20 +128,34 @@ private static void ProcessException(Activity activity, Exception ex)
...
+                    {
+                        activity.SetTag(attribute, property.GetValue(request));
+                    }
                 }
             }
+            catch (Exception)
+            {
+                // Guard against any reflection-related exceptions when running in AoT.
+                // See https://github.com/open-telemetry/opentelemetry-dotnet-contrib/issues/1543#issuecomment-1907667722.
+            }
```
Comment (martincostello) [inline src/OpenTelemetry.Instrumentation.AWS/Implementation/AWSTracingPipelineHandler.cs:158]: See https://github.com/open-telemetry/opentelemetry-dotnet-contrib/issues/1543#issuecomment-1907667722.

**Comment by martincostello** [comment]
FYI I've deployed a native AoT compiled .NET 8 application using this package to EKS and observed no issues related to this change and the native AoT compilation warning is no longer present.

File: test/OpenTelemetry.AotCompatibility.TestApp/OpenTelemetry.AotCompatibility.TestApp.csproj [hunk 2]
```diff
@@ -14,10 +14,10 @@
       .github\workflows\ci.yml so that it runs for the project being added.
```
Comment (Kielek) [inline test/OpenTelemetry.AotCompatibility.TestApp/OpenTelemetry.AotCompatibility.TestApp.csproj:14]: What's about ci.yml? Shouldn't it be updated?

Comment (martincostello) [reply test/OpenTelemetry.AotCompatibility.TestApp/OpenTelemetry.AotCompatibility.TestApp.csproj:14]: This library was already covered by the changes in https://github.com/open-telemetry/opentelemetry-dotnet-contrib/commit/397e9f8a2eba6fa2c72afac2d394a8698e4114cf.

**Review by Kielek — APPROVED** [review]

**Review by normj — APPROVED** [review]

**Comment by Kielek** [comment]
Merging as @srprash, agreed that @normj can act as component owner in https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/1539. There is only some technical requirements to merge mentioned PR.

If we speaking about release - please follow process from https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/CONTRIBUTING.md#how-to-request-for-release-of-package

**Comment by martincostello** [comment]
@Kielek Thanks - I've raised #1551, though I think the mentioned team is private as it hasn't tagged them due to my permissions.

**Pull request merged**